app/src/main/java/com/nervousfish/nervousfish/service_locator/NervousFish.java

```diff
@@ -0,0 +1,79 @@
+package com.nervousfish.nervousfish.service_locator;
+
+import android.app.Application;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+
+import com.nervousfish.nervousfish.modules.pairing.AndroidBluetoothService;
+import com.nervousfish.nervousfish.modules.pairing.PairingWrapper;
+
+/**
+ * The controller of the NervousFish application
+ */
+public final class NervousFish extends Application implements INervousFish {
+    private static NervousFish instance;
+
+    private AndroidBluetoothService bluetoothService;
+    private boolean bound;
+    private Runnable onServiceBoundRunnable;
+    private final ServiceConnection connection = new ServiceConnection() {
+        @Override
```

Comment:
Hmm, I'm starting to doubt if we should really use {@inheritDoc} everywhere. Intellij complains that it's cluttering the code because by default Java copies the javadoc of the superclass method already...